include/seqan3/alignment/pairwise/alignment_algorithm.hpp

```diff
@@ -149,448 +167,348 @@ class alignment_algorithm :
+                                              static_band const & band)
     {
-        assert(cfg_ptr != nullptr);
-
-        using std::get;
-
         static_assert(config_t::template exists<align_cfg::band>(),
                       "The band configuration is required for the banded alignment algorithm.");
 
-        auto const & band = get<align_cfg::band>(*cfg_ptr).value;
-
-        // ----------------------------------------------------------------------------
-        // Check valid band settings.
-        // ----------------------------------------------------------------------------
-
-        using diff_type = typename std::iterator_traits<std::ranges::iterator_t<first_range_t>>::difference_type;
+        using diff_type = std::iter_difference_t<std::ranges::iterator_t<fst_sequence_t>>;
         static_assert(std::is_signed_v<diff_type>,  "Only signed types can be used to test the band parameters.");
 
-        if (static_cast<diff_type>(band.lower_bound) > std::ranges::distance(first_range))
+        if (static_cast<diff_type>(band.lower_bound) > std::ranges::distance(fst_sequence))
         {
             throw invalid_alignment_configuration
             {
                 "Invalid band error: The lower bound excludes the whole alignment matrix."
             };
         }
 
-        if (static_cast<diff_type>(band.upper_bound) < -std::ranges::distance(second_range))
+        if (static_cast<diff_type>(band.upper_bound) < -std::ranges::distance(snd_sequence))
         {
             throw invalid_alignment_configuration
             {
                 "Invalid band error: The upper bound excludes the whole alignment matrix."
             };
         }
+    }
 
-        if constexpr (is_debug_mode)
-            initialise_debug_matrices(first_range, second_range);
+    /*!\brief Initialises the debug matrices for the given sequences.
+     * \tparam fst_sequence_t The type of the first sequence.
+     * \tparam snd_sequence_t The type of the second sequence.
+     *
+     * param[in] fst_sequence The first sequence.
+     * param[in] snd_sequence The second sequence.
+     *
+     * \details
+     *
+     * Initialises the debug matrices if the alignment algorithm is running in debug mode. See seqan3::align_cfg::debug
+     * for more information.
+     */
+    template <typename fst_sequence_t, typename snd_sequence_t>
+    constexpr void initialise_debug_matrices(fst_sequence_t & fst_sequence, snd_sequence_t & snd_sequence)
+    {
+        size_t rows = std::ranges::distance(snd_sequence) + 1;
+        size_t cols = std::ranges::distance(fst_sequence) + 1;
+
+        score_debug_matrix = score_debug_matrix_t{number_rows{rows}, number_cols{cols}};
+        trace_debug_matrix = trace_debug_matrix_t{number_rows{rows}, number_cols{cols}};
+    }
 
+    /*!\brief Compute the alignment by iterating over the alignment matrix in a column wise manner.
+     * \tparam index_t The type of the index of the current sequence pair.
```

Comment:
No, since these are private interfaces of the class we decided to not spoil them with template constraints. All required concepts must be checked by the public/protected interfaces. But no one can access from outside the private interfaces so they should be already safe within the class